babel/dates.py

```diff
@@ -1161,7 +1165,12 @@ def parse_date(string, locale=LC_TIME, format='medium'):
     if not numbers:
         raise ParseError("No numbers were found in input")
 
-    # TODO: try ISO format first?
+    # we try ISO-alike format first, meaning similar to YYYY-MM-DD
+    if len(numbers) == 3:
+        ints = list(map(int, numbers))
+        if ints[0] > 31 and ints[1] <= 12 and ints[2] <= 31:
```

Comment:
I think this could use a comment illuminating the idea behind this check in any case.

```suggestion
        # Only assume the date is an ISO 8601 date if
        # * the "year" component is >= 31 (so it can't be a day-of-month)
        # * the "month" component is in range
        # * the "day" component is in range
        if ints[0] > 31 and 1 <= ints[1] <= 12 and 1 <= ints[2] <= 31:
```

Reply:
The new regex and comment should make everything clear